motleycrew/agents/langchain/langchain.py

```diff
@@ -107,24 +91,30 @@ def materialize(self):
         if self.output_handler:
             self._agent.tools += [self._agent_finish_blocker_tool]
 
-            if isinstance(self._agent.agent, Runnable):
-                logger.info("Patching agent to force output via output handler")
-                agent_with_forced_output_via_handler = self._agent.agent | RunnableLambda(
-                    self._block_agent_finish
-                )
-                self._agent.agent = agent_with_forced_output_via_handler
-
-            elif isinstance(self._agent.agent, RunnableAgent):
-                logger.info("Patching agent to force output via output handler")
-                agent_with_forced_output_via_handler = self._agent.agent.runnable | RunnableLambda(
-                    self._block_agent_finish
-                )
-                self._agent.agent.runnable = agent_with_forced_output_via_handler
-
-            else:
-                logger.warning(
-                    "Agent is not a Runnable, cannot patch it to force output via output handler"
-                )
+            object.__setattr__(
+                self._agent.agent, "plan", self.agent_plan_decorator()(self._agent.agent.plan)
+            )
+
+            object.__setattr__(
+                self._agent,
+                "_take_next_step",
+                self.take_next_step_decorator()(self._agent._take_next_step),
```

Comment:
Let's make it `self.take_next_step_decorator(self._agent._take_next_step)`, no need for another function layer here.